components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -717,4 +722,30 @@ protected String sendRequest(String url, String accessToken) throws IOException
         }
         return builder.toString();
     }
+    
+    private String interpretQueryString (String queryString, Map<String,String[]> parameters) {
+	
+        String regex = "\\$\\{(\\w+)\\}";
+        Pattern pattern = Pattern.compile(regex);
```

Comment:
We could keep this in the class variable to improve performance, as the Pattern.compile is expensive operation and the regex never changes.
Also better to rename the regex to a meaningfull name

Reply:
Ok, both changes committed